test/functional/apm.test.js

```diff
@@ -701,6 +701,56 @@ describe('APM', function() {
     }
   });
 
+  // NODE-3358
+  describe('Internal state references', function() {
+    let client;
+
+    beforeEach(function(done) {
```

Comment:
Since `newClient` is synchronous, you can just pass a `function()` without a `done` argument and omit the the `done()` below.